src/main/java/duke/Storage.java

```
@@ -0,0 +1,86 @@
+    String s = currentRelativePath.toAbsolutePath().toString();
+
+    public Storage() {
+
+    }
+
+    public ArrayList<Task> load() throws DukeException {
+        try {
+            // System.out.println("Current absolute path is: " + s);
+            File dataDirectory = new File("data");
+            dataDirectory.mkdir();
+            File storage = new File("data/duke.txt");
+            storage.createNewFile();
+            // System.out.println(storage.getAbsolutePath());
+            Scanner savedTasks = new Scanner(storage);
+            ArrayList<Task> tasks = new ArrayList<>();
+
+            while (savedTasks.hasNextLine()) {
+                String curr = savedTasks.nextLine();
+
+                if (curr.startsWith("[T]")) {
+                    Task currentTask = new ToDo(curr.substring(7));
+                    tasks.add(currentTask);
+                } else if (curr.startsWith("[E]")) {
+                    int at = curr.lastIndexOf(" (at: ");
+                    int end = curr.lastIndexOf(")");
+                    Task currentTask = new Event(
+                            curr.substring(7, at),
+                            LocalDateTime.parse(curr.substring(at + 6, end),
+                                    DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm"))
+                    );
+                    tasks.add(currentTask);
+                } else if (curr.startsWith("[D]")) {
+                    int by = curr.lastIndexOf(" (by: ");
+                    int end = curr.lastIndexOf(")");
+                    Task currentTask = new Deadline(
+                            curr.substring(7, by),
+                            LocalDateTime.parse(curr.substring(by + 6, end),
+                                    DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm"))
+                    );
+                    tasks.add(currentTask);
+                } else {
+                    // do nothing
+                }
```

Is this best practice to have an empty else clause? Perhaps an exception could be thrown here

---

You are right, I should throw an exception here.